src/services/dsnp.ts

```diff
@@ -110,26 +115,35 @@ export const startSubscriptions = async (
   batchHandler: BatchAnnouncementHandler,
   registryHandler: RegistryUpdateHandler
 ): Promise<UnsubscribeFunction> => {
-  let blockNumber: number;
-  let blockIndex = 0;
-
   // subscribe to all announcements
+  let batchBlockNumber: number;
+  let batchBlockIndex = 0;
   const unsubscribeToBatchPublications = await core.contracts.subscription.subscribeToBatchPublications(
     (announcement: BatchPublicationLogData) => {
-      if (announcement.blockNumber !== blockNumber) {
-        blockNumber = announcement.blockNumber;
-        blockIndex = 0;
+      if (announcement.blockNumber !== batchBlockNumber) {
+        batchBlockNumber = announcement.blockNumber;
```

Comment:
Non-blocking: We should consider abstracting this since the logic is repeated below.

Reply:
This logic actually goes away now that we're getting transactionIndex from the SDK.